woocommerce/woocommerce-android
# [Coupons] Search for categories

Closes: #6362


Please make sure https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2437 is merged before merging this one.

### Description
This PR adds support for searching for product categories, and it adds unit tests for the main components of the screen.

### Testing instructions
1. Make sure coupons are enabled.
2. Open a coupon in the app.
3. Click on the menu button.
4. Click on edit coupon.
5. Click on "Add categories" or "Select categories".
6. Search for categories in the search view.
7. Confirm the results are retrieved from the API.
8. Confirm child categories are shown too when they are part of the results.

### Images/gif
<img width=400 src="https://user-images.githubusercontent.com/1657201/172811985-421f175f-eb3a-44bd-bb1b-acaaa76f3511.png"/>
